### PR TITLE
Add encryption between listener and processes

### PR DESCRIPTION
**Pull request by gregorjerse**

### CODE REVIEW

File: resolwe/flow/executors/zeromq_utils.py [hunk 1]
```diff
@@ -72,3 +73,20 @@ def __init__(
...
+    _instance = None
+
+    @classmethod
+    def has_instance(cls):
+        """Check if the instance exists."""
+        return cls._instance is not None
+
+    def __new__(cls):
+        """Overide new contructor to make it singleton."""
+        if cls._instance is None:
```
Comment (dblenkus) [inline resolwe/flow/executors/zeromq_utils.py]: Use `if not cls.has_instance():`.

File: resolwe/flow/managers/listener/listener.py [hunk 2]
```diff
@@ -560,11 +559,12 @@ def __init__(
         if zmq_socket is None:
             zmq_context: zmq.asyncio.Context = zmq.asyncio.Context.instance()
             # Start the authenticator.
-            auth = AsyncioAuthenticator(zmq_context)
-            auth.start()
-            auth.configure_curve_callback(
-                domain="*", credentials_provider=CurveCallback()
-            )
+            if ZMQAuthenticator.has_instance() is None:
```
Comment (dblenkus) [inline resolwe/flow/managers/listener/listener.py:622]: Why do you check whether the class was initialized? It defeats the purpose of making the class a singleton.

File: resolwe/flow/managers/workload_connectors/local.py [hunk 3]
```diff
@@ -35,5 +37,12 @@ def submit(self, data: Data, argv):
                 repr(argv),
             )
         )
+        process_environment = os.environ.copy()
+        print("LPK" * 10, LISTENER_PUBLIC_KEY)
```
Comment (dblenkus) [inline resolwe/flow/managers/workload_connectors/local.py:41]: Remove print.

Comment (gregorjerse) [reply resolwe/flow/managers/workload_connectors/local.py]: Fixed.

File: resolwe/flow/managers/listener/listener.py (hunk 2 above)
Comment (gregorjerse) [reply resolwe/flow/managers/listener/listener.py]: This was added since tests were failing due to the env bug. Removed.

**Review by dblenkus — APPROVED** [review]
Add a change log.

**Pull request merged**